js/foundation.offcanvas.js

```diff
@@ -50,13 +53,30 @@ class OffCanvas extends Plugin {
 
     this.$element.attr('aria-hidden', 'true');
 
-    this.$element.addClass(`is-transition-${this.options.transition}`);
+    this.$element.addClass(`is-transition-${this.options.transition} is-closed`);
 
     // Find triggers that affect this element and add aria-expanded to them
     this.$triggers = $(document)
       .find('[data-open="'+id+'"], [data-close="'+id+'"], [data-toggle="'+id+'"]')
       .attr('aria-expanded', 'false')
       .attr('aria-controls', id);
+    
+    // Get position by checking for related CSS class
+    this.position = this.$element.is('.position-left, .position-top, .position-right, .position-bottom') ? this.$element.attr('class').match(/position\-(left|top|right|bottom)/)[1] : this.position;
+
+    // Find off-canvas content, either by ID (if specified), by siblings or by closest selector (fallback)
+    if (this.options.contentId) {
+      this.$content = $('#'+this.options.contentId);
+    } else if (this.$element.siblings('[data-off-canvas-content]').length) {
+      this.$content = this.$element.siblings('[data-off-canvas-content]');
+    } else {
+      this.$content = this.$element.closest('[data-off-canvas-content]').first();
+    }
+
+    // Assume that the off-canvas element is nested if it isn't a sibling of the content
+    this.nested = this.$element.siblings('[data-off-canvas-content]').length && !this.options.contentId ? false : true;
```

Comment:
I'm using `this.nested` to emulate a fixed position via JS (see `_emulateFixedPosition()`) and force some options (e.g. no content scrolling) since this is necessary when nesting the off-canvas.

But it is quite difficult to determine if an off-canvas is nested – I don't want to parse the whole DOM for this!
That's why I assert the element is nested if it has no siblings content. But the usage of a contentId is a problem here since the related content element may be placed anywhere on the page (can be a sibling, can be nested, can be none of this).

Therefore I'm asserting that the off-canvas is nested if the contentId option is used. But that's obviously not a good solution as it limits the flexibility of usage.
I may ignore the contentId as it's probably not realistic to have several content elements on the same level and change it into:
`this.nested = !this.$element.siblings('[data-off-canvas-content]').length;`

What do you think?